### PR TITLE
AR: Refactor world scale

### PR DESCRIPTION
**Pull request by des12437**
Refactors https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/pull/466.

### CODE REVIEW

**Review by yo1995 — COMMENTED** [review]
Also do a auto re-indent for `WorldScaleGeoTrackingSceneView`

File: Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift [hunk 1]
```diff
@@ -123,34 +124,49 @@ public struct WorldScaleSceneView: View {
...
-                    statusText = "Acquiring current location."
-                }
-                try await locationDatasSource.start()
+                try await locationDataSource.start()
                 await withTaskGroup(of: Void.self) { group in
                     group.addTask {
-                        for await location in locationDatasSource.locations {
-                            await updateSceneView(for: location)
+                        for await location in locationDataSource.locations {
+                            self.lastLocationTimestamp = Date().timeIntervalSinceNow
```
[the remark this answers is not in the document]
Comment (des12437) [reply Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift:135]: Yes good catch, the location timestamp should be used instead.

File: Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift [hunk 2]
```diff
@@ -123,34 +124,49 @@ public struct WorldScaleSceneView: View {
...
+                                await updateSceneView(for: location, heading: heading)
+                                self.currentLocation = location
+                            }
                         }
                     }
                 }
-            } catch {
-                withAnimation {
-                    statusText = "Failed to acquire current location."
+            } catch {}
```
[the remark this answers is not in the document]
Comment (des12437) [reply Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift:143]: Yes, error handling can be presented in the status view.

File: Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift [hunk 3]
```diff
@@ -187,7 +203,7 @@ public struct WorldScaleSceneView: View {
     func shouldUpdateCamera(for location: Location) -> Bool {
         // Do not update unless the horizontal accuracy is less than a threshold.
         guard let currentCamera,
-              location.horizontalAccuracy < 5,
+              location.horizontalAccuracy < 45,
```
[the remark this answers is not in the document]
Comment (des12437) [reply Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift]: I removed this check and added checks to verify that the accuracy is greater than zero which indicates that the value is valid. Further accuracy checks will be addressed in `swift/issues/4875`.

**Review by yo1995 — APPROVED** [review]

**Pull request merged**